### PR TITLE
Remove call to componentWillReceiveProps

### PR DESCRIPTION
**Pull request by daggmano**
Issue #163, see https://fb.me/react-async-component-lifecycle-hooks

### CODE REVIEW

File: src/AutosizeInput.js [hunk 1]
```diff
@@ -57,13 +57,11 @@ class AutosizeInput extends Component {
 		this.copyInputStyles();
 		this.updateInputWidth();
 	}
-	componentWillReceiveProps (nextProps) {
-		const { id } = nextProps;
+	componentDidUpdate (prevProps, prevState) {
+		const { id } = prevProps;
 		if (id !== this.props.id) {
-			this.setState({ inputId: id || generateId() });
+			this.setState({ inputId: this.props.id || generateId() });
```
Comment (vadimka123) [inline src/AutosizeInput.js]: More correct to use `getDerivedStateFromProps` for update state based on props

**Comment by pbalenzuela** [comment]
any updates on this?

File: src/AutosizeInput.js (hunk 1 above)
Comment (pbalenzuela) [reply src/AutosizeInput.js]: Performance difference is despicable. Can't we move ahead and approve the PR? Using this deprecated event is much worse than this detail

**Comment by 0xdevalias** [comment]
Any updates on this?

**Comment by Methuselah96** [comment]
I've made [a fork](https://github.com/react-select-oss/react-input-autosize) with the fix for this released as [`react-input-autosize-oss@2.2.3`](https://github.com/react-select-oss/react-input-autosize/blob/master/HISTORY.md#v223--2020-11-12).

**Comment by Methuselah96** [comment]
Created alternative PR that uses `getDerivedStateFromProps` (https://github.com/JedWatson/react-input-autosize/pull/177).

**Comment by JedWatson** [comment]
We've merged #177 so this has been fixed, thanks for the PR @daggmano and sorry I didn't get to it sooner

**Pull request closed without merging**